crates/context/config/src/client.rs

```diff
@@ -0,0 +1,323 @@
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::convert::Infallible;
+use std::mem;
+
+use ed25519_dalek::Signature;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use thiserror::Error;
+
+use crate::repr::Repr;
+use crate::types::{self, Application, Capability, ContextId, ContextIdentity, Signed, SignerId};
+use crate::{ContextRequest, ContextRequestKind, Request, RequestKind};
+
+pub mod relayer;
+
+pub trait Transport {
+    type Error: std::error::Error;
+
+    #[allow(async_fn_in_trait)]
+    async fn send(
+        &self,
+        request: TransportRequest<'_>,
+        payload: Vec<u8>,
+    ) -> Result<Vec<u8>, Self::Error>;
+}
+
+#[derive(Debug)]
+pub struct TransportRequest<'a> {
+    pub network_id: Cow<'a, str>,
+    pub contract_id: Cow<'a, str>,
+    pub operation: Operation<'a>,
+}
```

Comment:
It would be nice to see Rustdocs added to new things 🙂

Reply:
We're the only consumers of these at this time, with no near-term plans of publishing them to docs.rs, and these pieces are under active development (maybe iterated on in successive PRs), documenting every iteration in code will only slow us down further, and I'm out next week.